### DISCUSSION TITLE
Is it possible to generate separate Logs in a Single Flow?

### DISCUSSION THREAD

**Discussion by shuylern** [jongpie/NebulaLogger#491] [Q&A] [answered]
Hello,

Is it possible to generate separate Log records in a single Flow?  I can create multiple Log Entries related back to a single Log currently. 

If we needed to generate **distinct** Logs with Log Entries via a single Flow transaction is that possible?

For instance we might want to isolate INFO logs for different events in the process (ex: an INFO Log for a part of the process and another distinct INFO Log for a separate part of the process).

Thanks in advance!  

Nathan

**Comment by jongpie** [comment, marked as answer]
Hi @shuylern - this is not possible in Nebula Logger, as a `Log__c` record always represents a single Apex transaction, so there's not a way to split log entries from the same transaction into different logs. But there might be some other options, depending on what you're trying to do. Can you clarify in your use case on why you want to isolate certain log entries? Is it for reporting purposes?

**Reply by shuylern** [reply]
@jongpie,

Thank you for the response.  I wanted to ensure I was not overlooking a feature before adding extra overhead.  

My use case example was:  an INFO Log for a part of the process and another distinct INFO Log for a separate part of the process.

I believe I can accomplish it by have issuing a Platform Event that incorporates the information needed to create a Log coupled with a Flow that listens for that specific Platform Event.  That Flow would then create an independent and isolated Log to match the use case.

Nathan

**Reply by jongpie** [reply]
@shuylern I still don't quite understand why you want to have multiple logs in this situation - splitting log entries (from the same transaction) into multiple logs isn't the intended use of Nebula Logger's data model. What does splitting the data into multiple logs provide you?

**Reply by shuylern** [reply]
@jongpie,

Having the ability to produce distinct Logs gives us the ability to track various items (see my use case above) and provide differing retention times for each.  We keep Logs of type X for 30 days and Logs of type Y for 180 days.  

It may be a different use of the data model but I always look for different ways to utilize good tools.  

Either way I do appreciate your creation and response to my initial question.

Thanks!

Nathan
